File: LeetCode/Simulation/Python/Leetcode1742-maximum-number-of-balls-in-a-box.py
class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        mp,ans = defaultdict(int), 0
        def calcu(x):
            res = 0
            while x:
                res += x % 10
                x //= 10
            return res
        for i in range(lowLimit, highLimit + 1):
            mp[calcu(i)] += 1
        for k,v in mp.items():
            ans = max(ans, v)
        return ans

File: LeetCode/Simulation/Rust/Leetcode1742-maximum-number-of-balls-in-a-box.rs
use std::collections::HashMap;
impl Solution {
    pub fn count_balls(low_limit: i32, high_limit: i32) -> i32 {
        let (mut ans, mut mp) = (0, HashMap::new());
        let calcu = |mut x| {
            let mut res = 0;
            while x != 0 {
                res += x % 10;
                x /= 10;
            }
            res
        };
        for i in low_limit..=high_limit {
            mp.entry(calcu(i)).and_modify(|e| {*e += 1}).or_insert(1);
        }
        for (k,v) in &mp {
            ans = ans.max(*v);
        }
        ans
    }
}